WooCommerce/Classes/ViewRelated/Products/Media/ProductImagesViewController.swift

```diff
@@ -208,6 +238,23 @@ private extension ProductImagesViewController {
         hasDeletedAnyImages = true
         productImageActionHandler.deleteProductImage(productImage)
     }
+
+    func handleProductImageStatusesReordering(_ reorderedProductImageStatuses: [ProductImageStatus]) {
+        // We have to validate if the order of the product images has been changed.
+        // The user can make some changes but the list may end up with the old order.
+        hasMovedAnyImages = validateIfAnyImagesHasMovedAfterReordering(reorderedProductImageStatuses)
+
+        productImageActionHandler.updateProductImageStatusesAfterReordering(reorderedProductImageStatuses)
+    }
+
+    /// Check if the given product images are in the same order as the original ones.
+    ///
+    func validateIfAnyImagesHasMovedAfterReordering(_ reorderedItems: [ProductImageStatus]) -> Bool {
+        let zippedProductImages = zip(reorderedItems, self.productImageStatuses)
```

Comment:
As mentioned earlier, I believe we should compare the reordered statuses with the original ones, i.e the ones sent to the controller initially. Since `productImageStatuses` is updated every time we reorder images, this will always return true.

But if we keep a reference to the initial statuses, there are more complications to be considered: uploading state, newly uploaded images.

If this takes too much effort, please feel free to file a separate issue for enhancement 👍